Mathlib/Topology/Order/LowerUpperTopology.lean

```diff
@@ -283,4 +471,30 @@ instance (priority := 90) LowerTopology.continuousInf : ContinuousInf α :=
   ⟨(infsInfHom : sInfHom (α × α) α).continuous⟩
 #align lower_topology.to_has_continuous_inf LowerTopology.continuousInf
 
-end CompleteLattice
+end CompleteLattice_LowerTopology
+
+section CompleteLattice_UpperTopology
+
+variable [CompleteLattice α] [CompleteLattice β] [TopologicalSpace α] [UpperTopology α]
+  [TopologicalSpace β] [UpperTopology β]
+
+protected theorem sSupHom.continuous (f : sSupHom α β) : Continuous f :=
+  sInfHom.continuous (α := αᵒᵈ) (β := βᵒᵈ) (sSupHom.dual.toFun f)
+
+-- see Note [lower instance priority]
+instance (priority := 90) UpperTopology.continuousInf : ContinuousSup α :=
+  ⟨(supsSupHom : sSupHom (α × α) α).continuous⟩
+
+end CompleteLattice_UpperTopology
+
+lemma UpperDual_iff_Lower [Preorder α] [TopologicalSpace α] :
+    UpperTopology αᵒᵈ ↔ LowerTopology α := by
+  constructor
+  · apply UpperTopology.instLowerTopologyOrderDualInstTopologicalSpaceOrderDualPreorder
+  · apply LowerTopology.instUpperTopologyOrderDualInstTopologicalSpaceOrderDualPreorder
```

Comment:
```suggestion
    constructor <;> intro _ <;> infer_instance
```
If that works. Same below

Reply:
Unfortunately
```
lemma UpperDual_iff_Lower [Preorder α] [TopologicalSpace α] :
    UpperTopology αᵒᵈ ↔ LowerTopology α := by
  constructor <;> intro _ <;> infer_instance
```
gives me
```
failed to synthesize instance
  LowerTopology α

```

Reply:
Ah right because it doesn't know how to go backwards. Then let's keep the current proof. Could you just give an explicit name to the instances so that it looks better?

Reply:
Done, thanks.